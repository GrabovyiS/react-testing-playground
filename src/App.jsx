import { useState } from "react";
import "./App.css";

const App = () => {
  const [heading, setHeading] = useState("Я тебя люблю");

  const handleClick = () => {
    setHeading("Давай встречаться");
  };

  return (
    <>
      <main>
        <h1>{heading}</h1>
        <button type="button" onClick={handleClick}>
          Change heading
        </button>
      </main>
    </>
  );
};

export default App;
