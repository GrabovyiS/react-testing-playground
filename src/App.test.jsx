import { render, screen } from "@testing-library/react";

import App from "./App";
import { expect } from "vitest";
import userEvent from "@testing-library/user-event";

describe("App", () => {
  // it("Correct snapshot on mount", () => {
  //   render(<App />);
  //   expect(screen.getByRole("main")).toMatchSnapshot();
  // });

  it("Correct counter after increment", () => {
    render(<App />);
    expect(screen.getByText("1")).toBeTruthy();
  });

  it("Correct counter after click", async () => {
    render(<App />);

    const user = userEvent.setup();
    const button = screen.getByText("click");

    await user.click(button);

    expect(screen.getByText("2")).toBeTruthy();
  });

  it("Correct heading after click", async () => {
    // Important note: things unmount after every test
    render(<App />);
    const user = userEvent.setup();
    const button = screen.getByRole("button");

    await user.click(button);

    const heading = screen.getByText("Давай встречаться");
    expect(heading).toBeInTheDocument();
  });
});
