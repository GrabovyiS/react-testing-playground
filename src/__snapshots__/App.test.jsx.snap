// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Correct snapshot on mount 1`] = `
<main>
  <h1>
    Я тебя люблю
  </h1>
  <button
    type="button"
  >
    Change heading
  </button>
</main>
`;
